pyomo/contrib/fbbt/interval.py

```diff
@@ -92,13 +101,111 @@ def power(xl, xu, yl, yu):
                 else:
                     lb = xl ** y
                     ub = xu ** y
+    elif yl == yu:
+        # the exponent is allowed to be fractional, so x must be positive
+        xl = 0
+        lb, ub = power(xl, xu, yl, yu)
     else:
-        lb = -math.inf
-        ub = math.inf
+        msg = 'encountered an exponent where the base is allowed to be negative '
+        msg += 'and the exponent is allowed to be fractional and is not fixed. '
+        msg += 'Assuming the lower bound of the base to be 0.'
+        warnings.warn(msg)
+        logger.warning(msg)
+        xl = 0
+        lb, ub = power(xl, xu, yl, yu)
 
     return lb, ub
 
 
+def _inverse_power1(zl, zu, yl, yu, orig_xl, orig_xu):
+    """
+    z = x**y => compute bounds on x
+    x = exp(ln(z) / y)
+    """
+    xl, xu = log(zl, zu)
+    xl, xu = div(xl, xu, yl, yu)
+    xl, xu = exp(xl, xu)
+
+    # if y is an integer, then x can be negative
+    if yl == yu and yl == round(yl):
```

Comment:
This is a pretty complicated if tree .. could you provide (terse, inline) comments giving hints at the conditions that each branch is trying to cover?

Reply:
Yes. I meant to add some docs but forgot.